### PR TITLE
test(kafka): Streamline polling of multiple messages

### PR DESCRIPTION
**Pull request by Dav1dde**
#skip-changelog

### CODE REVIEW

File: tests/integration/fixtures/processing.py [hunk 1]
```diff
@@ -167,6 +167,31 @@ def poll(self, timeout=None):
...
+        message = self.poll(timeout)
+        while message is not None:
+            yield message
+            messages += 1
+
+            if messages == n:
+                self.assert_empty()
+                break
+
+            message = self.poll(min(2, timeout) if n is None else timeout)
```
Comment (jjbayer) [inline tests/integration/fixtures/processing.py:194]: Can't we use `min(2, timeout)` even if `n is not None`?

Comment (jjbayer) [inline tests/integration/fixtures/processing.py:194]: Should the 2 be configurable? E.g. `def poll_many(self, initial_timeout=None, subsequent_timeout=2, n=None)`?

**Review by jjbayer — COMMENTED** [review]
Nice!

File: tests/integration/fixtures/processing.py (hunk 1 above)
Comment (Dav1dde) [reply tests/integration/fixtures/processing.py:194]: It makes everything a bit more fragile then, if you expect a certain amount of items and they are not there we can afford to wait a little bit longer (should never happen). But if for some reason something actually is a bit slower we won't fail with the longer timeout.

Comment (Dav1dde) [reply tests/integration/fixtures/processing.py:194]: Yes, I haven't done that yet because I couldn't decide on good names.

Comment (jjbayer) [reply tests/integration/fixtures/processing.py:194]: Makes sense.

Comment (jjbayer) [reply tests/integration/fixtures/processing.py:194]: nit: Can you put that reasoning in a code comment?

**Review by jjbayer — APPROVED** [review]

**Pull request merged**